### PR TITLE
Add ability to set destroy failed jobs on a per job basis

### PR DESCRIPTION
**Pull request by edgibbs**
We recently needed the ability to selectively delete failed jobs, even though our default is to keep all failed jobs.  I noticed there was a pull request (#742) but it has been sitting for a few months without tests.  I went ahead and built out the feature with tests included and I updated the README with an example.  DJ has been a real work horse for us for a number of years now, and I appreciate all the work you've done maintaining it.

### CODE REVIEW

**Comment by jdelStrother** [comment]
Funnily enough I discovered a need for this just yesterday.  Yes please - would be a useful feature.

**Comment by edgibbs** [comment]
Any update on when/if this will be merged?

**Comment by albus522** [comment]
Right now it does not cleanly merge. Also the first half of the worker changes are not needed and is better as is.

File: lib/delayed/worker.rb [hunk 1]
```diff
@@ -247,7 +247,7 @@ def failed(job)
           say "Error when running failure callback: #{error}", 'error'
           say error.backtrace.join("\n"), 'error'
         ensure
-          self.class.destroy_failed_jobs ? job.destroy : job.fail!
+          destroy_failed_jobs(job) ? job.destroy : job.fail!
```
Comment (albus522) [inline lib/delayed/worker.rb]: Lets simplify this. Lets make the call `destroy_failed_jobs?` and just alway call `job.destroy_failed_jobs`.

File: lib/delayed/backend/base.rb [hunk 2]
```diff
@@ -139,6 +139,10 @@ def max_run_time
         end
       end
 
+      def destroy_failed_jobs
+        payload_object.destroy_failed_jobs if payload_object.respond_to?(:destroy_failed_jobs)
+      end
+
```
Comment (albus522) [inline lib/delayed/backend/base.rb]: Then here, update the method to return `Delayed::Worker.destroy_failed_jobs` if the payload does not respond to `destroy_failed_jobs`

Comment (albus522) [inline lib/delayed/backend/base.rb]: I am also thinking we make the payload object implement `destroy_failed_jobs?` instead of `destroy_failed_jobs`.

**Comment by edgibbs** [comment]
Thanks for the feedback, I'm running to work now, but I'll make the changes tonight.

**Comment by edgibbs** [comment]
I think I got all the changes in from the comments.  I went ahead and updated the method name to destroy_failed_jobs?.  The only issue I see there is it has a slightly different name from the Delayed::Worker.destroy_failed_jobs class attribute.

**Pull request merged**